### PR TITLE
Implement single tap touchscreen

### PR DESCRIPTION
**Pull request by emmauss**
This implements single tap touchscreen. Tested with libtransistor hid tests. I also fixed the incorrect Hid_Num_Entries constant.
Previous branch had commit errors

### CODE REVIEW

File: Ryujinx.Core/Hid.cs [hunk 1]
```diff
@@ -170,16 +169,62 @@ public void Init(long HidOffset)
...
+                    NumTouches = 1
+                },
+                Touches = new HidTouchScreenEntryTouch[16]
+            };
+
+            //Only supports single touch
+            hidTouchScreenEntry.Touches[0] = TouchPoint;
+
+            Marshal.StructureToPtr(hidTouchScreenEntry, HidPtr, false);
+
```
Comment (gdkchan) [inline Ryujinx.Core/Hid.cs]: Those new lines on the end of the method are not necessary.

File: Ryujinx.Core/Hid/HidTouchScreen.cs [hunk 2]
```diff
@@ -51,4 +51,5 @@ public struct HidTouchScreen
         [MarshalAs(UnmanagedType.ByValArray, SizeConst = 0x3C0)]
         public byte[] Padding;
     }
+
```
Comment (gdkchan) [inline Ryujinx.Core/Hid/HidTouchScreen.cs]: New line not necessary.

File: Ryujinx/Properties/launchSettings.json [hunk 3]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (gdkchan) [inline Ryujinx/Properties/launchSettings.json]: Why this file was added? Looks like something that was used for testing and shouldn't be included on the PR.

File: Ryujinx/Ui/GLScreen.cs [hunk 4]
```diff
@@ -346,6 +347,30 @@ protected override void OnUpdateFrame(FrameEventArgs e)
...
+                        Y = (uint)Mouse.Y,
+
+                        //Placeholder values till more data is acquired
+                        DiameterX = 10,
+                        DiameterY = 10,
+                        Angle = 90,
+
+                        //Only support single touch
+                        TouchIndex = 0,
+
```
Comment (gdkchan) [inline Ryujinx/Ui/GLScreen.cs]: New line not necessary.

File: Ryujinx/Properties/launchSettings.json (hunk 3 above)
Comment (emmauss) [reply Ryujinx/Properties/launchSettings.json]: dont know. looks like it was from me switching between debug and release in visual studio,  or from doing something in vs that is not code specific

**Comment by emmauss** [comment]
visual studio did that. I closed a brace and it aligned everything there

**Comment by gdkchan** [comment]
You can undo it by pression the undo button or using Ctrl+Z when visual studio tries to auto-format the file for you. Also I think this can be disabled in settings, althrough I never really tried and I don't use visual studio anymore.

Thanks for the contribution!

**Pull request merged**